### DISCUSSION TITLE
자바의 디스크립터에 대한 고찰

### DISCUSSION THREAD

**Discussion by ahngilwoong** [Study-2-Modern-Java-In-Action/Study-2-Modern-Java-in-Action#31] [CHAPTER 3 람다 표현식 - Q&A] [answered]
이번 챕터의 학습으로
자바의 디스크립터는 함수형 프로그래밍을 지원하면서도 기존 자바 코드와의 호환성을 유지할 수 있다는 장점이 있다고 파악했습니다.
하지만 결정적으로 다른 함수형 프로그래밍 언어들과의 차이가 있다고 생각합니다.
다른 함수형 프로그래밍 언어를 잘 아시는 분들이 있다면 자바의 디스크립터와 비교하며 자바 디스크립터의 단점들을 말씀해주시면 좋을 것 같습니다.

**Comment by corock** [comment, marked as answer]
다른 함수형 프로그래밍 언어는 잘 모르지만, 자바에서 제공하는 함수 디스크립터는 추상 메서드를 1개만 정의할 수 있다는 점이 제약인 듯 합니다.

예를 들어, 자바스크립트의 경우 아래처럼 한 객체 내에 여러 메서드를 포함할 수 있습니다.

```javascript
const calculator = {
  add: (a, b) => { return a + b; },
  subtract: (a, b) => { return a - b; },
  multiply: (a, b) => { return a * b; },
  divide: (a, b) => { return a / b; },
};
```
